editor/sidebar/post-visibility/index.js

```diff
@@ -106,35 +106,61 @@ class PostVisibility extends Component {
 		return (
 			<div className="editor-post-visibility">
 				<span>{ __( 'Visibility' ) }</span>
-				<button className="editor-post-visibility__toggle button-link" onClick={ this.toggleDialog }>
+				<button
+					type="button"
+					aria-expanded={ this.state.opened }
+					className="editor-post-visibility__toggle button-link"
+					onClick={ this.toggleDialog }
+				>
 					{ getVisibilityLabel( visibility ) }
 				</button>
 
 				{ this.state.opened &&
 					<div className="editor-post-visibility__dialog">
 						<div className="editor-post-visibility__dialog-arrow" />
-						<div className="editor-post-visibility__dialog-legend">
-							{ __( 'Post Visibility' ) }
-						</div>
-						{ visibilityOptions.map( ( { value, label, info, onSelect, checked } ) => (
-							<label key={ value } className="editor-post-visibility__dialog-label">
-								<input
-									type="radio"
-									value={ value }
-									onChange={ onSelect }
-									checked={ checked } />
-								{ label }
-								{ <div className="editor-post-visibility__dialog-info">{ info }</div> }
-							</label>
-						) ) }
+						<fieldset>
+							<legend className="editor-post-visibility__dialog-legend">
+								{ __( 'Post Visibility' ) }
+							</legend>
+							{ visibilityOptions.map( ( { value, label, info, onSelect, checked } ) => (
+								<div key={ value } className="editor-post-visibility__choice">
+									<input
+										type="radio"
+										name="editor-post-visibility__setting"
```

Comment:
It's not likely we'd have more than one of these components on the page, but given that we're injecting `instanceId` anyways, maybe we should include it in the input name?